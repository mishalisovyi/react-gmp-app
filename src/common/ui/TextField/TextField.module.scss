@use '../../../colors';
@use '../../../mixins';

.TextField {
  @include mixins.typography-body;
  @include mixins.font-montserrat-regular;

  background-color: colors.$black-panther;
  padding: 16px 20px;
  border-radius: 4px;
  font-size: 18px;
  border: none;
}