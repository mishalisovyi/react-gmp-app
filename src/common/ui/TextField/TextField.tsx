import React, { forwardRef } from 'react';
import { ComponentWithRefProps } from 'common/interfaces';
import styles from './TextField.module.scss';

interface TextFieldProps extends ComponentWithRefProps {
  placeholder?: string,
  value?: string,
}

const TextField = forwardRef(
  ({ placeholder, value }: TextFieldProps, ref: React.ForwardedRef<HTMLInputElement>) => {
    return (
      <input type="text" ref={ref} defaultValue={value} placeholder={placeholder} className={styles['TextField']} />
    );
  },
);

TextField.defaultProps = {
  placeholder: '',
  value: '',
};

export { TextField };
