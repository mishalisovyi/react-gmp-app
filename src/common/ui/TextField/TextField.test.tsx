import React from 'react';
import renderer from 'react-test-renderer';

import { TextField } from './TextField';

it('should render text field', () => {
  const tree = renderer.create(<TextField value="test value" placeholder="test placeholder" />);

  expect(tree.toJSON()).toMatchSnapshot();
});
