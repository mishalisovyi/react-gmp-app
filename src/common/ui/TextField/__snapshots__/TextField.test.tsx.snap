// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render text field 1`] = `
<input
  defaultValue="test value"
  placeholder="test placeholder"
  type="text"
/>
`;
